src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptor.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Compilation.TagHelpers
+{
+    /// <summary>
+    /// A metadata class describing a required tag helper attribute.
+    /// </summary>
+    public class TagHelperRequiredAttributeDescriptor
+    {
+        /// <summary>
+        /// Supported CSS value operators.
+        /// </summary>
+        public static readonly char[] SupportedCSSValueOperators = { '=', '^', '$' };
+
+        /// <summary>
+        /// The HTML attribute name.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// The HTML attribute selector value. If <see cref="IsCssSelector"/> is not <c>true</c>, this field is
+        /// ignored.
+        /// </summary>
+        public string Value { get; set; }
+
+        /// <summary>
+        /// An operator that modifies how a required attribute is applied to an HTML attribute value or name.
+        /// </summary>
+        public char Operator { get; set; }
+
+        /// <summary>
+        /// Indicates if the <see cref="TagHelperRequiredAttributeDescriptor"/> represents a CSS selector.
+        /// </summary>
+        public bool IsCssSelector { get; set; }
```

Comment:
There's still something off about this descriptor. `IsCssSelector` isn't useful except to slightly reduce the number of times `string.Equals()` is called in `IsMatch()` and describes something about the source that isn't important later. Meanwhile `Operator` is _not_ the operator in a CSS selector but an odd thing that controls handling of both the attribute name and value and is only understandable reading the `IsMatch()` source.

We need an actual descriptor here, not exposure of the implementation details. For example, `bool IsPrefixMatch` controlling how the attribute name is handled and `char Operator` or `SomeEnum CssValueComparison` controlling only how the attribute value is handled.